3rd-party/cove/app/mainform.cpp

```diff
@@ -892,6 +842,31 @@ void mainForm::on_applyFIRFilterPushButton_clicked()
 	if (!newImageBitmap.isNull()) imageBitmap = newImageBitmap;
 	ui.imageView->setImage(&imageBitmap);
 }
+
+/* \class mainForm::BwBitmapUndoStep
+ * \brief Embedded struct holding undo data.
+ */
+
+mainForm::BwBitmapUndoStep::BwBitmapUndoStep(mainForm* form, QImage image, bool vectorizable)
+    : formPtr { form }
+    , image { image }
+    , suitableForVectorization { vectorizable }
+{
+	// nothing
+}
+
+void mainForm::BwBitmapUndoStep::redo()
+{
+	undo(); // the same data swap
+}
+
+void mainForm::BwBitmapUndoStep::undo()
+{
+	Q_ASSERT(formPtr);
```

Comment:
While it is not quite Qt-ish, I try to use references when I mean "not null" in new code. (I know Mapper is full of non-zero pointers...)

Reply:
Good catch. I was pulled into the use of pointer under the API. A reference is indeed the right tool. Thanks!